### PR TITLE
node compatibility #140 - Add file parsing support

### PR DESCRIPTION
**Pull request by bmcorum**
Supports CSV files with either a csv or txt extension. Supports single file or array of files. When passing an array of files you can pass in an override config ( { file: filename, config: overrideConfig } ).

Examples

parseFiles(file1.csv)

config = { ... }
parseFiles(file1.csv, config)

parseFiles([file1.csv, file2.csv], config)

overrideConfig = { ... }
parseFiles([{ file: file1.csv, config: overrideConfig}, file2.csv], config)

### CODE REVIEW

File: papaparse.js [hunk 1]
```diff
@@ -167,6 +168,42 @@
...
+			});
+			return results;
+		} else {
+			var results = {
+				data: [],
+				errors: []
+			};
+			if ((/(\.csv|\.txt)$/).test(_input)) {
+				try {
+					var contents = fs.readFileSync(_input).toString();
```
Comment (mholt) [inline papaparse.js]: How well does this work with really large files?

File: papaparse.js [hunk 2]
```diff
@@ -167,6 +168,42 @@
...
+				else
+					results.push(ParseFiles(input, _config));
+			});
+			return results;
+		} else {
+			var results = {
+				data: [],
+				errors: []
+			};
+			if ((/(\.csv|\.txt)$/).test(_input)) {
```
Comment (mholt) [inline papaparse.js]: Let's not restrict the file extension - there could be any number of different valid ones, or none at all.

File: papaparse.js [hunk 3]
```diff
@@ -167,6 +168,42 @@
...
+					return results;
+				}
+			} else {
+				results.errors.push({
+					type: '',
+					code: '',
+					message: 'Unsupported file type.',
+					row: ''
+				});
+				return results;
```
Comment (mholt) [inline papaparse.js]: @bmcorum Again, I don't think we need to worry about this. But good for thinking of it :+1:

File: papaparse.js (hunk 1 above)
Comment (bluej100) [inline papaparse.js]: Yeah, readFileSync seems like the wrong call here (see http://stackoverflow.com/questions/13822085/fs-readfilesync-seems-faster-than-fs-readfile-is-it-ok-to-use-for-a-web-app-in for perspective from someone who does more Node than I do). I would suggest that you write a ChunkStreamer subclass (like FileStreamer) so you get the benefit of all of Papa's chunked reads and async processing.

**Comment by bmcorum** [comment]
@bluej100  I would need to wrap the ParseFiles in a promise which would not be terribly difficult. I will need to find the time to write and test the code.

@mholt Not sure how large a "large" file would be but I have a few that are a few hundred lines long (are we talking 100s of KB, 100s of MB, or ***I hope not 100s of GB). I am currently using Baby since Papa is a bit bloated for strict Node use but I can understand using Papa when trying to target multiple platforms.

**Comment by pokoli** [comment]
I'm closing this in favour of #370

**Pull request closed without merging**